File: python/program/WildAnimal.py

from Animal import Animal

class WildAnimal(Animal):
    def __init__(self, species="", name="", habitat="", endangered=False):
        super().__init__(species, name)
        self.__habitat = habitat
        self.__endangered = endangered
    
    def set_habitat(self, habitat):
        self.__habitat = habitat
    
    def set_endangered(self, endangered):
        self.__endangered = endangered
    
    def get_habitat(self):
        return self.__habitat
    
    def is_endangered(self):
        return self.__endangered

File: java/program/WildAnimal.java

public class WildAnimal extends Animal {
    private String habitat;
    private boolean endangered;
    
    public WildAnimal() {
    }
    
    public WildAnimal(String species, String name, String habitat, boolean endangered) {
        super(species, name);
        this.habitat = habitat;
        this.endangered = endangered;
    }
    
    public void setHabitat(String habitat) {
        this.habitat = habitat;
    }
    
    public void setEndangered(boolean endangered) {
        this.endangered = endangered;
    }
    
    public String getHabitat() {
        return this.habitat;
    }
    
    public boolean isEndangered() {
        return this.endangered;
    }
}